File: pset6/caesar/caesar.py
from cs50 import get_string
import sys

if len(sys.argv) != 2:
    print("Please enter 1 command line argument")
    exit(1)

key = int(sys.argv[1])
s = get_string("plaintext: ")

print("ciphertext: ", end="")

for i in s:
    if i.isalpha():
        if i.islower():
            j = chr(((ord(i) - ord('a') + key) % 26) + ord('a'))
            print(j, end="")
        elif i.isupper():
            k = chr(((ord(i) - ord('A') + key) % 26) + ord('A'))
            print(k, end="")
    else:
        print(i, end="")

print()

File: pset2/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    //ensure a command line arugment is entered
    if (argc != 2)
    {
        printf("Please enter a key and plaintext to be coded\n");
        return 1;
    }
    else
    {
        string s = get_string("plaintext: ");
        //changing the users input for the key to an int
        string num = argv[1];
        int key = atoi(num);

        printf("ciphertext: ");
        for (int i = 0, n = strlen(s); i < n; i++)
        {
            if (isalpha(s[i]))
            {
                //maintain lowercase
                if (islower(s[i]))
                {
                    //shift character by the key
                    printf("%c", (((s[i] + key) - 97) % 26) + 97);
                }
                //maintain uppercase
                else if (isupper(s[i]))
                {
                    //shit character by the key
                    printf("%c", (((s[i] + key) - 65) % 26) + 65);
                }
            }
            else //don't alter non-alphabetical characters
            {
                printf("%c", s[i]);
            }
        }
        printf("\n");
    }
    return 0;
}